r/vignettes/developing.Rmd

```diff
@@ -60,36 +71,38 @@ brew install apache-arrow
 brew install apache-arrow --HEAD
 ```
 
+### Windows and Linux
+
 On Windows and Linux, you can download a .zip file with the arrow dependencies from the
 nightly repository.
-Windows users then can set the `RWINLIB_LOCAL` environment variable to point to that
-zip file before installing the `arrow` R package. On Linux, you'll need to create a `libarrow` directory inside the R package directory and unzip that file into it. Version numbers in that
-repository correspond to dates, and you will likely want the most recent.
 
 To see what nightlies are available, you can use Arrow's (or any other S3 client's) S3 listing functionality to see what is in the bucket `s3://arrow-r-nightly/libarrow/bin`:
 
 ```
 nightly <- s3_bucket("arrow-r-nightly")
 nightly$ls("libarrow/bin")
 ```
+Version numbers in that repository correspond to dates.
 
-## Developer environment setup
+#### Windows
 
-If you need to alter both the Arrow C++ library and the R package code, or if you can’t get a binary version of the latest C++ library elsewhere, you’ll need to build it from source too. This section discusses how to set up a C++ build configured to work with the R package. For more general resources, see the [Arrow C++ developer
-guide](https://arrow.apache.org/docs/developers/cpp/building.html).
+Windows users then can set the `RWINLIB_LOCAL` environment variable to point to the zip file containing the arrow dependencies before installing the arrow R package. 
+
+#### Linux
+
+On Linux, you'll need to create a `libarrow` directory inside the R package directory and unzip the zip file containing the arrow dependencies into it. 
+
+## R and C++
 
-There are four major steps to the process — the first three are relevant to all Arrow developers, and the last one is specific to the R bindings:
+If you need to alter both the Arrow C++ library and the R package code, or if you can't get a binary version of the latest C++ library elsewhere, you'll need to build it from source. This section discusses how to set up a C++ build configured to work with the R package. For more general resources, see the [Arrow C++ developer guide](https://arrow.apache.org/docs/developers/cpp/building.html).
 
-1. Configuring the Arrow library build (using `cmake`) — this specifies how you want the build to go, what features to include, etc.
-2. Building the Arrow library — this actually compiles the Arrow library
-3. Install the Arrow library — this organizes and moves the compiled Arrow library files into the location specified in the configuration
-4. Building the R package — this builds the C++ code in the R package, and installs the R package for you
+There are five major steps to the process — the first four are relevant to all Arrow developers, and the last one is specific to developers making changes to the R package.
 
-### Install dependencies {.tabset}
+### Step 1 - Install dependencies
```

Comment:
Did you intend to get rid of `{.tabset}` here? It's a bit funky how it works (and is very much non-standard/hacked on to pkgdown), but it gives us a nice tabbed interface for interacting with the setup steps (e.g. https://arrow.apache.org/docs/r/articles/developing.html#install-dependencies).

Reply:
Yeah, my intention was to make it really easy to navigate through the page, using the table of contents that appears at the side, and the heading levels, to signpost what bits applies to what.

I could add it back in though - will see how including it affects the contents list.

Reply:
If the goal is to keep the tab titles in the table of contents, it's possible we could do that by changing some of the javascript that [we manipulate to make this whole thing work](https://github.com/apache/arrow/blob/master/r/pkgdown/extra.js). Something funny that we do there is that we load [the javascript for sticky tabs](https://cdn.jsdelivr.net/gh/rstudio/rmarkdown@47d837d3d9cd5e8e212b05767454f058db7d2789/inst/rmd/h/navigation-1.1/tabsets.js), and then alter some of the functions to do the things we need them to do in-place.